projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -25,6 +25,35 @@ import useSocialMediaConnections from '../../hooks/use-social-media-connections'
 import VideoPreview from '../video-preview';
 import styles from './styles.module.scss';
 
+/**
+ * Get meta data from a VideoPress video.
+ *
+ * @param {object} video - VideoPress media object.
+ * @returns {{mime: string, fileSize: number, length: number}} - Metadata
+ */
+const getVideoPressMetadata = async video => {
+	if (
+		! video?.media_details?.videopress?.original ||
+		! video?.media_details?.videopress?.duration
+	) {
+		return {};
+	}
+
+	const response = await fetch( video?.media_details?.videopress?.original, { method: 'HEAD' } );
```

Comment:
> Is there any chance that this request would get blocked by adblockers etc.?

That's not a problem thankfully! I think it would be great if we could get this information from the server, but doing it in the client is fine for now.